src/course-unit/context/iFrameContext.tsx

```diff
@@ -0,0 +1,43 @@
+import React, {
+  createContext, MutableRefObject, useRef, useCallback, useMemo, ReactNode,
+} from 'react';
+
+export interface IframeContextType {
+  setIframeRef: (ref: MutableRefObject<HTMLIFrameElement | null>) => void;
+  sendMessageToIframe: (messageType: string, payload: unknown) => void;
+}
+
+export const IframeContext = createContext<IframeContextType | undefined>(undefined);
+
+export const IframeProvider: React.FC = ({ children }: { children: ReactNode }) => {
+  const iframeRef = useRef<HTMLIFrameElement | null>(null);
+  const setIframeRef = useCallback((ref: MutableRefObject<HTMLIFrameElement | null>) => {
+    iframeRef.current = ref.current;
+  }, []);
+
+  const sendMessageToIframe = useCallback((messageType: string, payload: any) => {
+    const iframeWindow = iframeRef?.current?.contentWindow;
+    if (iframeWindow) {
+      try {
+        iframeWindow.postMessage({ type: messageType, payload }, '*');
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Failed to send message to iframe:', error);
```

Comment:
Removed

Reply:
The idea to not use `console()` is to avoid showing ugly messages to the end user.   I'm not sure if throwing a new `Error()` will get managed properly, but I suspect it won't.

How about if we used [logError](https://github.com/openedx/frontend-platform/blob/07ee86cf54d26dd1a8a98e24cf1a402e98157f61/src/logging/interface.js#L70) instead?  Right now it only works with newRelic, but at least it takes case of both sides of the issue: not showing the message to the end user while still allowing the operator to know about it.